### DISCUSSION TITLE
Obtaining the first frame

### DISCUSSION THREAD

**Discussion by Smokedhammer** [ultralytics/yolov5#10671] [Q&A]
Hi!

Is there a way I can take just the first frame from the webcam feed and apply detection on it? Basically, I don't want live detection, but the detection on the first frame that the webcam sees.

**Comment by alijafari79** [comment]
> Hi!
> 
> Is there a way I can take just the first frame from the webcam feed and apply detection on it? Basically, I don't want live detection, but the detection on the first frame that the webcam sees.

Hi Dear @Smokedhammer , 

If you mean to get the first frame of your webcam, I'll say yes you can !

You may use OpenCV to access the first frame of the webcam by using the code below :

```
import cv2
 
f = cv2.VideoCapture(0)
 
rval, frame = f.read()
```
Now the frame variable from the above code is the first frame of your webcam .....

Change 0 in  cv2.VideoCapture() to 1 if you want to use the webcam other than the main webcam on laptop or PC.


If my answer helped you please mark it as answer !

Thanks.

**Reply by Smokedhammer** [reply]
Hey,

thanks for the response. I understand what you're trying to get at, but where would I find this in the YOLO v5 directories. I cant find something similar.

Also how can I only take the first frame of the webcam and shutoff the webcam afterward? I basically just need an image (aka the first frame of the video from the webcam)

**Reply by alijafari79** [reply]
> Hey,
> 
> thanks for the response. I understand what you're trying to get at, but where would I find this in the YOLO v5 directories. I cant find something similar.
> 
> Also how can I only take the first frame of the webcam and shutoff the webcam afterward? I basically just need an image (aka the first frame of the video from the webcam)

Hi Dear @Smokedhammer

Opencv is a python package you need to install it using :

```
pip install opencv-python
```

You can use this python package to just prepare your input data for yolov5, I mean just use this to get the first frame and then save the first frame as an image by adding : 

```
cv2.imwrite(filename, frame) 
```
to the end of the code in the previous  message.

Then you can close the camera after taking the first frame by adding these two more lines to the code :

```
f.release()
cv2.destroyAllWindows()
``` 
Now you have the image saved in your directory, feel free to use it with Yolov5.

If you find my answer helpful please check the "Mark as answer" button :)

Thanks.
